tools/istio-iptables/pkg/builder/iptables_builder_impl.go

```diff
@@ -87,7 +87,16 @@ func (rb *IptablesBuilderImpl) AppendRuleV6(chain string, table string, params .
 
 func (rb *IptablesBuilderImpl) BuildV4() []string {
 	rules := []string{}
+	chainTableLookupMap := make(map[string]bool)
 	for _, r := range rb.rules.rulesv4 {
+		chainTable := fmt.Sprintf("%s:%s", r.chain, r.table)
+		// Create new chain if key: `chainTable` isn't present in map
+		if _, present := chainTableLookupMap[chainTable]; !present {
+			cmd := fmt.Sprint(strings.Join(
+				[]string{dependencies.IPTABLES, "-t", r.table, "-N", r.chain}, " "))
+			rules = append(rules, cmd)
+			chainTableLookupMap[chainTable] = true
+		}
```

Comment:
I would sort the rules after the loop exits, so that the chains are always created in a predictable order.
```
sort.Strings(rules)
```

Reply:
Not sure if I want to `sort`; given that order matters for iptables?

Reply:
It matters for rules, yes, but it doesn't matter for chain creations. And at this point, `rules` only contains commands to create chains.

Reply:
As we discussed offline, please do a first loop over the rules before this loop, which only add commands to create the chains, then sort those chain creation commands, then do this loop to append all the rules.
So there should be 2 loops in this method.